### PR TITLE
Build afni from source

### PR DESCRIPTION
**Pull request by giovtorres**
Added some bits to build AFNI from source.  It may require some refactoring.  To use it, you would have to pass these flags simultaneously: `use_binaries=false build_source=true`(see [here](https://github.com/giovtorres/neurodocker/blob/build_afni_source/neurodocker/interfaces/afni.py#L69-L72)).  Also, `version` in this context refers to the git branch, so `latest` is not valid.  How should this interface handle these?

Thanks.

/cc @Shotgunosine

### CODE REVIEW

**Comment by kaczmarj** [comment]
Thanks @giovtorres. I will test out the build soon.

If the user requests `latest`, we should infer `master`. I included [something similar in the ANTs interface](https://github.com/kaczmarj/neurodocker/blob/master/neurodocker/interfaces/ants.py#L125-L126).

File: neurodocker/interfaces/afni.py [hunk 1]
```diff
@@ -161,3 +186,62 @@ def install_binaries(self):
...
+                "\n&& perl -p -i -e 's/^USE_LOCAL_X_TREE/#USE_LOCAL_X_TREE/' Makefile"
+                "\n&& perl -p -i -e 's/XLIBS = \$\(XROOT\)\/lib64\/libXm.a -lXt/XLIBS = \$\(XROOT\)\/lib64\/libXm.a \$\(XROOT\)\/lib\/x86_64-linux-gnu\/libXm.a -lXt/' Makefile"
+                "\n&& perl -p -i -e 's/^# XLIBS =/XLIBS =/' Makefile"
+                "\n&& perl -p -i -e 's/^CCOLD.*/CCOLD  = \$\(CC\)/' Makefile"
+                "\n&& perl -p -i -e 's/(^LFLAGS.*)/$1 -L\/usr\/lib\/x86_64-linux-gnu/' Makefile"
+                "\n&& perl -p -i -e 's/(^PLFLAGS.*)/$1 -L\/usr\/lib -L\/usr\/lib\/x86_64-linux-gnu/' Makefile"
+                "\n&& perl -p -i -e 's/-lXpm -lXext/-lXpm -lfontconfig -lXext/' Makefile"
+                "\n&& perl -p -i -e 's/(^SUMA_INCLUDE_PATH.*)/$1 -I\/usr\/lib\/x86_64-linux-gnu\/glib-2.0\/include/' Makefile"
+                "\n&& make INSTALLDIR=/opt/afni vastness"
+                "".format(self.version, **manage_pkgs[self.pkg_manager]))
```
Comment (kaczmarj) [inline neurodocker/interfaces/afni.py]: Can you include some logic to user `master` branch if the user requests to build from the latest version?

File: neurodocker/interfaces/afni.py [hunk 2]
```diff
@@ -161,3 +186,62 @@ def install_binaries(self):
...
+    def install_from_source(self):
+        """Return Dockerfile instructions to download and build AFNI from
+        source."""
+        pkgs = self._get_source_dependencies()
+
+        cmd = ("{install}"
+               "".format(**manage_pkgs[self.pkg_manager]).format(pkgs=pkgs))
+
+        if self.pkg_manager == "yum":
+            cmd += ("\n&& yum install -y -q R R-devel")
```
Comment (kaczmarj) [inline neurodocker/interfaces/afni.py]: Can `R` and `R-devel` be added to the list of `yum` dependencies in `_get_source_dependencies()`?

File: neurodocker/interfaces/afni.py [hunk 3]
```diff
@@ -161,3 +186,62 @@ def install_binaries(self):
...
+            cmd += ("\n&& /opt/afni/rPkgsInstall -pkgs ALL"
+                    "\n&& rm -rf /tmp/*")
+
+        cmd = indent("RUN", cmd)
+
+        env_plugin_path = "AFNI_PLUGINPATH=/opt/afni"
+        env_plugin_path = indent("ENV", env_plugin_path)
+
+        env_cmd = "PATH=/opt/afni:$PATH"
+        env_cmd = indent("ENV", env_cmd)
```
Comment (kaczmarj) [inline neurodocker/interfaces/afni.py]: Can you merge `ENV AFNI_PLUGINPATH=/opt/afni` and `ENV PATH=/opt/afni:$PATH` into one `ENV` layer?

**Comment by kaczmarj** [comment]
hi @giovtorres - can you merge with current master? also, the build works for me on debian stretch and centos 7.

File: neurodocker/interfaces/afni.py (hunk 2 above)
Comment (giovtorres) [reply neurodocker/interfaces/afni.py]: The `R` and `R-devel` packages are found in the EPEL repository.  The `epel-release` package is installed in the yum source dependency list, and therefore both R packages need to come after in a separate `yum` invocation.

File: neurodocker/interfaces/afni.py (hunk 1 above)
Comment (giovtorres) [reply neurodocker/interfaces/afni.py]: Done.

File: neurodocker/interfaces/afni.py (hunk 3 above)
Comment (giovtorres) [reply neurodocker/interfaces/afni.py]: Good idea. Done.

**Comment by kaczmarj** [comment]
Thank you @giovtorres.

**Pull request merged**